HW 10 sprint

AndrewkaW opened the pull request.

AndrewkaW committed:
Подключил библиотеку Glide, Уведомил ОС об использовании в приложении интернета.

AndrewkaW committed:
Замокал список треков

AndrewkaW committed:
TEST COMMIT

AndrewkaW committed:
Сверстал элемент track_item_view.xml

AndrewkaW committed:
Добвил элемент RecyclerView в activity_search.xml , и реавлизовал его в коде. Отканил настройки приложенитя для корректного запуска. Подправил верстку track_item_view.xml

AndrewkaW committed:
Сделал что бы Inflate выполнялся в конструкторе ViewHolder.

SeveNChaK commented on app/src/main/java/com/practicum/playlistmaker/Track.kt:
```diff
@@ -0,0 +1,53 @@
+        Glide.with(itemView)
+            .load(model.artworkUrl100)
+            .placeholder(R.drawable.default_art_work)
+            .transform(RoundedCorners(2))
```
⚠️ Здесь радиус задается в пикселях. Поэтому значение из макета (которое мы берем как значение в `dp`) необходимо преобразовать в пиксели.
Удобнее всего будет вынести это значение в ресурсы, а потом достать его здесь: `itemView.resources.getDimensionPixelSize(нужный ресурс)`

SeveNChaK commented on app/src/main/res/layout/track_item_view.xml:
```diff
@@ -0,0 +1,65 @@
+                android:id="@+id/artistNameText"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                tools:text="Имя артиста"/>
```
⚠️ Длинное имя артиста выталкивает время
<img width="454" alt="image" src="https://user-images.githubusercontent.com/22057568/224622256-06c13c11-b06b-4965-bc1c-883714096d49.png">
Можно добавить аттрибут `android:layout_weight="1"` и сделать `android:layout_width="0dp"`, тогда должно починиться

SeveNChaK reviewed (commented):
Не надо добавлять папку `.idea` к отслеживанию в `git`. Она у каждого своя и генерируется при создании/клонирование проекта на устройстве.

⏫ У `Toolbar` есть специальный метод, чтобы отслеживать клики именно на кнопку навигации (в нашем случае это стрелка "назад"). Лучше использовать его, чтобы не слушать клики по всему тулбару `setNavigationOnClickListener`

AndrewkaW committed:
Изменил слушателя на тул барах в SearchActivity.kt и SettingsActivity.kt.
Исправил всерстку track_item_view.xml. Теперь имя артиста не выдавливает время трека. Вынес значени радиуса закругления обложки в ресурсы, ссылка используеться в коде.

SeveNChaK commented on app/src/main/res/values/dimens.xml:
```diff
@@ -13,5 +13,6 @@
 
     <dimen name="art_work_size">45dp</dimen>
     <dimen name="artist_time_text_size">10sp</dimen>
+    <dimen name="art_work_radius">2dp</dimen>
```
⚠️ Ресурс сделал, но не используешь его в коде в месте, где задаешь радиус

AndrewkaW replied:
@SeveNChaK Не знаю как так вышло. Изменения файла трек не попали в коммит. Может галку случайно снял.(( 
Глупо максимально...

AndrewkaW committed:
Вынес значени радиуса закругления обложки в ресурсы, ссылка используеться в коде.
